### PR TITLE
Update apollo to v3

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because it bumps apollo to v3.

**PR intended to be tested with API branch:** main

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { ListSettings, ListViews, Pagination } from "./types";
 
 export const APP_MOUNT_URI = process.env.APP_MOUNT_URI || "/";
 export const APP_DEFAULT_URI = "/";
-export const API_URI = process.env.API_URI;
+export const API_URI = "https://master.staging.saleor.cloud/graphql/";
```
Comment (orzechdev) [inline src/config.ts:7]: Is it intended?

File: src/fragments/collections.ts [hunk 2]
```diff
@@ -37,8 +37,8 @@ export const collectionDetailsFragment = gql`
 
 // This fragment is used to make sure that product's fields that are returned
 // are always the same - fixes apollo cache
-// https://github.com/apollographql/apollo-client/issues/2496
-// https://github.com/apollographql/apollo-client/issues/3468
+// https://github.com/apollographql/@apollo/client/issues/2496
+// https://github.com/apollographql/@apollo/client/issues/3468
```
Comment (orzechdev) [inline src/fragments/collections.ts:41]: It breaks links in comments xd

File: package.json [hunk 3]
```diff
@@ -36,10 +36,7 @@
     "@uiw/react-color-material": "^0.1.0",
     "@uiw/react-color-saturation": "0.0.34",
     "apollo": "^2.32.5",
-    "apollo-cache-inmemory": "^1.6.5",
-    "apollo-client": "^2.6.8",
     "apollo-client-preset": "^1.0.6",
-    "apollo-link": "^1.2.12",
     "apollo-link-batch-http": "^1.2.12",
```
Comment (orzechdev) [inline package.json:40]: This might be also removed due to the usage of `@apollo/client/link/batch-http`

File: src/config.ts (hunk 1 above)
Comment (dominik-zeglen) [reply src/config.ts:7]: Eh huge diffs, ofc not intended

File: package.json [hunk 4]
```diff
@@ -36,10 +36,7 @@
     "@uiw/react-color-material": "^0.1.0",
     "@uiw/react-color-saturation": "0.0.34",
     "apollo": "^2.32.5",
```
Comment (jwm0) [inline package.json:38]: What about this one?

File: package.json [hunk 5]
```diff
@@ -36,10 +36,7 @@
...
-    "apollo-cache-inmemory": "^1.6.5",
-    "apollo-client": "^2.6.8",
     "apollo-client-preset": "^1.0.6",
```
Comment (jwm0) [inline package.json:39]: and this one

File: package.json [hunk 6]
```diff
@@ -36,10 +36,7 @@
...
-    "apollo-link": "^1.2.12",
     "apollo-link-batch-http": "^1.2.12",
     "apollo-link-context": "^1.0.18",
```
Comment (jwm0) [inline package.json:41]: I don't think this one is used as well

File: package.json (hunk 4 above)
Comment (dominik-zeglen) [reply package.json:38]: Newest version is `2.33`

**Review by orzechdev — APPROVED** [review]

File: package.json (hunk 4 above)
Comment (jwm0) [reply package.json]: Are we still using `apollo` package though?

**Review by jwm0 — APPROVED** [review]

**Comment by michalina-graczyk** [comment]
QA approved

**Review by michalina-graczyk — APPROVED** [review]

**Pull request merged**